src/main/scan.ts

```diff
@@ -0,0 +1,285 @@
+type Range = GoogleAppsScript.Spreadsheet.Range;
+type TextStyle = GoogleAppsScript.Spreadsheet.TextStyle;
+type RichTextValue = GoogleAppsScript.Spreadsheet.RichTextValue;
+
+// RegEx pattern used to identify daily tabs
+const DAILY_TAB_REGEX_PATTERN: RegExp = /^(MON|TUE|WED|THU|FRI|SAT|SUN) ([1-9]|1[0-2])\/([1-9]|[12][0-9]|3[01])$/;
+
+// Column indices for event rows
+const DATE_ROW_INDEX: number = 0;
+const DATE_COL_INDEX: number = 0;
+const START_TIME_COL_INDEX: number = 0;
+const END_TIME_COL_INDEX: number = 1;
+const WHO_COL_INDEX: number = 2;
+const NUM_ATTENDEES_COL_INDEX: number = 3;
+const WHAT_COL_INDEX: number = 4;
+const WHERE_COL_INDEX: number = 5;
+const IN_CHARGE_COL_INDEX: number = 6;
+const HELPERS_COL_INDEX: number = 7;
+const FOOD_LEAD_COL_INDEX: number = 8;
+const CHILDCARE_COL_INDEX: number = 9;
+const NOTES_COL_INDEX: number = 10;
+
+// Number of minutes in an hour
+const MIN_IN_HOUR: number = 60;
+
+interface CellData {
+    readonly value: any,
+    readonly linkUrl: string | null,
+    readonly strikethrough: boolean
+}
+
+/**
+ * Fetches all of the tabs of the active spreadsheet
+ *
+ * @returns an array of spreadsheet tabs
+ */
+export function getAllSpreadsheetTabs(): Sheet[] {
+    const spreadsheet: Spreadsheet = SpreadsheetApp.getActiveSpreadsheet();
+    return spreadsheet.getSheets();
+}
+
+/**
+ * Fetches all of the active, non-hidden, daily tabs from an array of spreadsheet tabs
+ * 
+ * @param spreadsheetTabs array of spreadsheet tabs to search through
+ * @returns array of active daily tabs
+ */
+export function getActiveDailyTabs(spreadsheetTabs: Sheet[]): Sheet[] {
+    const dailyActiveTabs: Sheet[] = [];
+    for(const tab of spreadsheetTabs) {
+        checkIfActiveDailyTab(tab, dailyActiveTabs);
+    }
+
+    return dailyActiveTabs;
+}
+
+/**
+ * Fetches the rows from a spreadsheet tab that correspond to an actual event on the Onestop
+ * 
+ * @param spreadsheetTab spreadsheet tab to search through
+ * @returns an array of Row objects
+ */
+export function getRowsWithEvents(spreadsheetTab: Sheet): Row[] {
+    const dataRange: Range = spreadsheetTab.getDataRange();
+    const cellData = getCellData(dataRange);
+    const currentDate = getDateOfDailyTab(cellData);
+
+    const rows: Row[] = [];
+    for(let i = 0; i < cellData.length; i++) {
+        const rowData: CellData[] = cellData[i];
+        processRow(rowData, rows, currentDate);
+    }
+
+    return rows;
+}
+
+/**
+ * Checks if a given tab is not hidden and is a daily tab. If it is, add it to the dailyActiveTabs output array
+ * 
+ * @param tab the tab to check
+ * @param dailyActiveTabs output array to add the tab to if it meets the criteria of not being hidden and being a daily tab
+ */
+function checkIfActiveDailyTab(tab: Sheet, dailyActiveTabs: Sheet[]): void {
```

Comment:
in general let's try to avoid output arguments; it makes readability a bit harder. in this case it should be an easier fix since your return type currently is void, so it can be replaced by just doing a boolean check if it's an active daily tab

Reply:
Output parameter removed. Function was updated to just return a boolean representing whether the given tab is an active and daily tab